### PR TITLE
Release 3.7 merge to master

### PR DESCRIPTION
**Pull request by wubin01**
**What does this PR do?**
Release 3.7 merge to master

**Why are these changes required?**

**This PR has been tested by:**
- Unit Tests
- Manual Testing

**Follow up**

**Extra details**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -54,4 +54,366 @@ nodeId.properties
 Wallet
 
 # vm_trace
-/vm_trace/
\ No newline at end of file
+/vm_trace/
+framework/note_encdec_test/account/000005.log
```
Comment (andelf) [inline .gitignore]: should use glob pattern /*

**Comment by wubin01** [comment]
> * fix .gitignore
> * manage your removed code with version control instead of block comments.

solve in the next version

**Comment by andelf** [comment]
> * fix .gitignore
> * manage your removed code with version control instead of block comments.

At least, the `.gitignore` file is not acceptable.

**Comment by Balance2020Happiness2021** [comment]
> # [Codecov](https://codecov.io/gh/tronprotocol/java-tron/pull/3043?src=pr&el=h1) Report
> > Merging [#3043](https://codecov.io/gh/tronprotocol/java-tron/pull/3043?src=pr&el=desc) into [master](https://codecov.io/gh/tronprotocol/java-tron/commit/89dfa18548a327e8094562f9366939747878e9bb&el=desc) will **not change** coverage by `%`.
> > The diff coverage is `n/a`.
> 
> [Impacted file tree graph](https://codecov.io/gh/tronprotocol/java-tron/pull/3043?src=pr&el=tree)
> 
> ```diff
> @@            Coverage Diff            @@
> ##             master    #3043   +/-   ##
> =========================================
>   Coverage     31.09%   31.09%           
>   Complexity     1942     1942           
> =========================================
>   Files           386      386           
>   Lines         19453    19453           
>   Branches       2240     2240           
> =========================================
>   Hits           6049     6049           
>   Misses        12695    12695           
>   Partials        709      709           
> ```
> 
> 
> 
> ------
> 
> [Continue to review full report at Codecov](https://codecov.io/gh/tronprotocol/java-tron/pull/3043?src=pr&el=continue).
> > **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> > `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> > Powered by [Codecov](https://codecov.io/gh/tronprotocol/java-tron/pull/3043?src=pr&el=footer). Last update [89dfa18...89dfa18](https://codecov.io/gh/tronprotocol/java-tron/pull/3043?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).
>

**Pull request merged**